### ISSUE TITLE
[a11y]: <TableExpandRow>'s specified id for the ARIA property 'aria-controls' value is not valid

### ISSUE THREAD

**Issue by jinhua-ibm** [carbon-design-system/carbon#18808]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.76.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker Version 4.0.0

### Description

As the checker results, it says the id of each <TableExpandedRow> has violation against aria controls. 

![Image](https://github.com/user-attachments/assets/f9e716cc-64de-4efe-8d0f-861f2140085d)

### Reproduction/example

none

### Steps to reproduce

Create a carbon <Datatable> with <TableExpandedRow> and perform a11y checker scan.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by 2nikhiltom** [comment]
Hi @jinhua-ibm I could not create this on out storybook
could you recreate this on [sandbox](https://github.com/carbon-design-system/sandboxes)

**Comment by jinhua-ibm** [comment]
> Hi [@jinhua-ibm](https://github.com/jinhua-ibm) I could not create this on out storybook could you recreate this on [sandbox](https://github.com/carbon-design-system/sandboxes)

Hey @2nikhiltom , try this story https://react.carbondesignsystem.com/?path=/docs/components-datatable-expansion--overview. (But not sure the code reflects what it presents)

**Comment by annawen1** [comment]
@jinhua-ibm I ran the accessibility checker on the story you mentioned: https://react.carbondesignsystem.com/iframe.html?globals=&args=&id=components-datatable-expansion--default&viewMode=story and don't see the `aria-controls` violation 

<img width="1782" alt="Image" src="https://github.com/user-attachments/assets/841da402-9550-4c57-b742-a6edf1840811" />

**Comment by jinhua-ibm** [comment]
Is it being fixed? And I had checkboxes too, not sure if that would add the a11y violations?

**Comment by annawen1** [comment]
@jinhua-ibm Can you share a working example with stackblitz or on sandbox that shows this issue? We need a working example in order for us to debug and fix. As mentioned above, we aren't able to see this issue appearing on our end.

**Comment by jinhua-ibm** [comment]
@annawen1 I’m swamped with other priorities at the moment and haven’t been able to investigate this thoroughly. Since this is an a11y issue, could you help look into it further?

**Comment by jinhua-ibm** [comment]
Resolved this by properly configuring TableExpandedRow props and code structure, closing
